<?xml version="1.0"?>
<!--
Software License Agreement (BSD)

\file      reconstruction.launch
\authors   Pengdi Huang <alualu628628@163.com>
\copyright Copyright (c) 2022, 05, All rights reserved.


THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WAR-
RANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, IN-
DIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->

<launch>

    <arg name="sf_output_path" default="" />

    <!-- use slam transfer -->
    <!-- <arg name="cloudintopic" default="/slam_points" /> -->
    <!-- <arg name="odomintopic" default="/slam_odom" /> -->

    <!-- use aloam -->
    <arg name="cloudintopic" default="/velodyne_cloud_registered" />
    <arg name="odomintopic" default="/aft_mapped_to_init" />

    <!-- lidar type -->
    <!-- in-outdoor -->
    <arg name="lidar_line_min" default="0"/>
    <arg name="lidar_line_max" default="15"/>
    <!-- new shool -->
    <!-- <arg name="lidar_line_min" default="1"/>
    <arg name="lidar_line_max" default="46"/> -->
    <!-- kitti -->
    <!-- <arg name="lidar_line_min" default="0"/>
    <arg name="lidar_line_max" default="50"/> -->
    <!-- maicity -->
    <!-- <arg name="lidar_line_min" default="-1"/>
    <arg name="lidar_line_max" default="52"/> -->

    <arg name="cloudoutopic" default="/frame_cloudnormals" />

    <arg name="polygonoutopic" default="/frame_meshs" />

    <arg name="mesh_algo_topic" default="/frame_mesh_algo" />
    <arg name="mesh_algo_view_topic" default="/frame_mesh_algo_view" />

    <arg name="outcloudtfid" default="map"/>

    <arg name="polygontfid" default="map"/>

    <arg name="samplepcframenum"  default="1" />

    <!-- step value in “for” point cloud,  larger means fewer points-->
    <arg name="sampleinpnum"  default="4" />

    <arg name="sectornum"  default="12" />

    <arg name="pointzoffset" default="1.2"/> 

    <arg name="useghpr" default="true"/>

    <arg name="multi_thread" default="true"/>


    <!-- simple reconstruction (Adjacency method)-->
    <arg name="lidartype" default="16"/>
    <arg name="showconfidence" default="false"/>
    <arg name="combine_clouds" default="false"/>

    <!-- frame_reconstruction node in frame_reconstruction pkg-->
    <node name="simple_frame" pkg="simple_frame" type="simple_frame" output="screen">

        <!-- parameters-->	

        <param name="file_output_path" type="string" value="$(arg sf_output_path)"  />

        <!-- define input topic name-->
		<param name="cloud_in_topic" type="string" value="$(arg cloudintopic)"  />

		<param name="odom_in_topic" type="string" value="$(arg odomintopic)" />

        <!-- define output topic name-->
        <param name="cloud_out_topic" type="string" value="$(arg cloudoutopic)" />

        <param name="polygon_out_topic" type="string" value="$(arg polygonoutopic)" />

        <param name="mesh_algo_topic" type="string" value="$(arg mesh_algo_topic)" />
        <param name="mesh_alog_view_topic" type="string" value="$(arg mesh_algo_view_topic)" />
		
        <!-- frame id related-->
	    <param name="sample_pcframe_num" type="int" value="$(arg samplepcframenum)" />

        <!-- sampling number-->
	    <param name="sample_inputpoints_num"  type="int" value="$(arg sampleinpnum)" />

        <param name="sector_num"  type="int" value="$(arg sectornum)" />

	    <!--param name="odom_zoffset" type="double" value="$(arg odomzoffset)" /-->

        <param name="viewp_zoffset" type="double" value="$(arg pointzoffset)"/>

        <!-- tf id related-->
        <param name="outcloud_tf_id" type="string" value="$(arg outcloudtfid)" />

        <param name="polygon_tf_id" type="string" value="$(arg polygontfid)" />

        <param name="lidar_type" type="int" value="$(arg lidartype)" />
        <param name="lidar_line_min" type="int" value="$(arg lidar_line_min)" />
        <param name="lidar_line_max" type="int" value="$(arg lidar_line_max)" />

        <param name="use_ghpr" type="bool" value="$(arg useghpr)" />

        <param name="show_confidence" type="bool" value="$(arg showconfidence)" />

        <param name="combine_clouds" type="bool" value="$(arg combine_clouds)" />

        <param name="multi_thread" type="bool" value="$(arg multi_thread)" />

	</node>


</launch>


